Add aymanbagabas planck keyboard map

aymanbagabas opened the pull request:
Add aymanbagabas planck keymap. Based on Planck via with encoder and audio support.

<!--- Provide a general summary of your changes in the title above. -->

<!--- This template is entirely optional and can be removed, but is here to help both you and us. -->
<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->



## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation


## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

elpekenin commented on keyboards/planck/keymaps/aymanbagabas/olkb-planck-rev6.1-aymanbagabas.json:
Looks like a VIA's JSON, doesnt belong to QMK's repo

aymanbagabas replied:
Removed and added a link to config in the readme.

waffle87 reviewed (commented):
Aside from Encoder Map, how does this PR differ from the [existing Planck VIA keymap](https://github.com/qmk/qmk_firmware/tree/master/keyboards/planck/keymaps/via)? Audio is already enabled at the keyboard level, and bootmagic is conditionally enabled when VIA is enabled. (also the flag you have for bootmagic is incorrect, will not have enabled bootmagic, and will have caused a compilation error...)

aymanbagabas commented:
> Aside from Encoder Map, how does this PR differ from the [existing Planck VIA keymap](https://github.com/qmk/qmk_firmware/tree/master/keyboards/planck/keymaps/via)? Audio is already enabled at the keyboard level, and bootmagic is conditionally enabled when VIA is enabled. (also the flag you have for bootmagic is incorrect, will not have enabled bootmagic, and will have caused a compilation error...)

You're absolutely right. I removed those rules and updated the config. Other than the Encoder Map and the Encoder config, nothing differs from the existing Planck VIA keymap.

waffle87 commented:
The preferred approach would be to simply add encoder map functionality to the existing VIA keymap then.

aymanbagabas commented:
> The preferred approach would be to simply add encoder map functionality to the existing VIA keymap then.

The encoder map can be different on a case-to-case basis, some people have two encoders, and some have more.

drashna commented on keyboards/planck/keymaps/aymanbagabas/config.h:
```diff
@@ -0,0 +1,49 @@
+// Most tactile encoders have detents every 4 stages
+#define ENCODER_DIRECTION_FLIP
+
+#define ENCODER_RESOLUTION 4
```
This is the default 
```suggestion
```

zvecr commented:
Unfortunately since this PRs creation, QMK has [changed its stance](https://docs.qmk.fm/#/ChangeLog/20230827?id=notable-changes) on accepting user keymaps to the repository.

> qmk/qmk_firmware is no longer accepting PRs for keymaps other than for manufacturer-supported keymaps. User keymap workflow has been documented [here](https://docs.qmk.fm/#/newbs) for several years. This change is to progressively reduce the maintenance burden on the project, and to allow us to focus on the core features of QMK.

In the near future we will be providing a more user focused workflow for keymaps, but till then the suggestion is to keep changes in your own fork.

While we appreciate the time that was taken on this PR, hopefully you understand we are closing it, so that QMK can improve in the long term.
